registry/client/repository.go

```diff
@@ -163,67 +158,156 @@ func (r *repository) Signatures() distribution.SignatureService {
 	}
 }
 
-type signatures struct {
-	manifests distribution.ManifestService
-}
-
-func (s *signatures) Get(dgst digest.Digest) ([][]byte, error) {
-	m, err := s.manifests.Get(dgst)
-	if err != nil {
-		return nil, err
+func (r *repository) Tags(ctx context.Context) distribution.TagService {
+	return &tags{
+		client:  r.client,
+		ub:      r.ub,
+		context: r.context,
+		name:    r.Name(),
 	}
-	return m.Signatures()
 }
 
-func (s *signatures) Put(dgst digest.Digest, signatures ...[]byte) error {
-	panic("not implemented")
-}
-
-type manifests struct {
-	name   string
-	ub     *v2.URLBuilder
-	client *http.Client
-	etags  map[string]string
+// tags implements remote tagging operations.
+type tags struct {
```

Comment:
Seeing as this is no longer repository, can we start another file in the `client` package?